File: Going Dutch/MCCurrency.m
//
//  MCCurrency.m
//  We all pay
//
//  Created by Mark Cornelisse on 23/07/14.
//  Copyright (c) 2014 Mark Cornelisse. All rights reserved.
//

#import "MCCurrency.h"
#import "MCExchangeRate.h"
#import "MCPayment.h"
#import "MCSharedBill.h"

@implementation MCCurrency

#pragma mark - NSManagedObject

- (void)awakeFromInsert {
    [super awakeFromInsert];
    [self setPrimitiveValue:NSUUID.UUID.UUIDString forKey:@"uniqueID"];
    NSDate *now = NSDate.date;
    [self setPrimitiveValue:now forKey:@"dateCreated"];
    [self setPrimitiveValue:now forKey:@"dateModified"];
    
}
#pragma mark - NSObject

@end

File: Going Dutch/MCCurrency.swift
//
//  MCCurrency.swift
//  We all pay
//
//  Created by Mark Cornelisse on 26/08/2025.
//  Copyright © 2025 Mark Cornelisse. All rights reserved.
//

import Foundation
import CoreData

@objc(MCCurrency) public final class MCCurrency: NSManagedObject {

    // MARK: NSManagedObject
    
    public override func awakeFromInsert() {
        super.awakeFromInsert()
        
        self.setPrimitiveValue(UUID().uuidString, forKey: "uniqueID")
        let now = Date()
        self.setPrimitiveValue(now, forKey: "dateCreated")
        self.setPrimitiveValue(now, forKey: "dateModified")
    }
    
    // MARK: NSObject
}
